agent/acs/updater/updater.go

```diff
@@ -79,7 +80,7 @@ func NewUpdater(cfg *config.Config, state dockerstate.TaskEngineState, dataClien
 	taskEngine engine.TaskEngine) *updater {
 	return &updater{
 		config:     cfg,
-		httpclient: httpclient.New(updateDownloadTimeout, false),
+		httpclient: httpclient.New(updateDownloadTimeout, false, version.String(), config.OSType),
```

Comment:
nit - this may give out the impression that we are referring to http version (since this is an http client), so consider using a more specific package identifier like `agentVersion` to avoid confusion.

Same for the other occurrences

Reply:
Good point, thanks for calling this subtle point out! I will revise accordingly.